### PR TITLE
Feature/199 parametrize persistance xml

### PR DESCRIPTION
**Pull request by michalbabiarz**
This commit includes:
- db module fixes - mostly AbstractDao
- unit and integration tests introduced
- parametrization od db conncetion for MySql via environment.csv
- some unit test fixes in core

Test results:
http://de-muc-devonfw-allure01:8888/job/test/job/mrchecker-database-module/job/feature%252F199-parametrize-persistance-xml/59/

http://de-muc-devonfw-allure01:8888/job/test/job/mrchecker-core-module/job/feature%252F199-parametrize-persistance-xml/9/

### CODE REVIEW

File: mrchecker-framework-modules/mrchecker-database-module/src/resources/settings.properties [hunk 1]
```diff
@@ -1 +1,2 @@
-database.lib=./lib/dbdrivers
\ No newline at end of file
+database.lib=./lib/dbdrivers
+core.isEncryptionEnabled=true
```
Comment (MalgorzataDzienia) [inline mrchecker-framework-modules/mrchecker-database-module/src/resources/settings.properties:2]: encription  true or false? I don't see you are using encryption, yet

File: mrchecker-framework-modules/mrchecker-database-module/src/test/java/com/capgemini/mrchecker/database/unit/base/environment/EnvironmentParamTest.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+		{
+			put("DEV", "Vx4hylOJhNzRc5kKm4gr");
+			put("QA", "pass");
+		}
+	};
+
+	private static final Map<String, String> connectionUrlEnvMapping = new HashMap<String, String>(){
+		{
+			put("DEV", "localhost:3307/int_tests");
+			put("QA", "de-muc-devonfw-allure01:3306/int_tests");
```
Comment (MalgorzataDzienia) [inline mrchecker-framework-modules/mrchecker-database-module/src/test/java/com/capgemini/mrchecker/database/unit/base/environment/EnvironmentParamTest.java]: do we want to propagate our url in github?

**Review by MalgorzataDzienia — APPROVED** [review]

File: mrchecker-framework-modules/mrchecker-database-module/src/test/java/com/capgemini/mrchecker/database/unit/base/environment/EnvironmentParamTest.java (hunk 2 above)
Comment (michalbabiarz) [reply mrchecker-framework-modules/mrchecker-database-module/src/test/java/com/capgemini/mrchecker/database/unit/base/environment/EnvironmentParamTest.java]: Fixed to get host from System.properties

File: mrchecker-framework-modules/mrchecker-database-module/src/resources/settings.properties (hunk 1 above)
Comment (michalbabiarz) [reply mrchecker-framework-modules/mrchecker-database-module/src/resources/settings.properties:2]: Someone might want to use the encryption in future.

**Pull request merged**